Blank page when no vocabulary items are found

speedysera opened the pull request:
This pull request addresses issue #2273 - when filtering by language, there is no indication why the page is blank when there are no vocabulary items found.

jiru commented on src/Template/Vocabulary/add_sentences.ctp:
```diff
@@ -55,6 +60,13 @@ $this->set('title_for_layout', $this->Pages->formatTitle($title));
+        <div class="empty-info-text">
+            <?php
+                if ($vocabulary->count() == 0) {
+                    echo "There are no requests. ";
```
This string include an unnecessary space at the end. It should also made translatable using `__("your translatable string here")`, otherwise it will stay in English regardless of the UI language.

speedysera replied:
thanks, learned something new about php today!

jiru commented on src/Template/Vocabulary/add_sentences.ctp:
```diff
@@ -55,6 +60,13 @@ $this->set('title_for_layout', $this->Pages->formatTitle($title));
             )
             ?>
         </div>
+        <div class="empty-info-text">
```
When there is at least one request, this div will show up as empty, potentially creating a blank space. I think it would be better to make it *not show up at all* by putting it *inside* the `if()` branch.

speedysera replied:
good point!

jiru commented on webroot/css/layouts/default.css:
```diff
@@ -795,4 +795,11 @@ html[dir="rtl"] .md-button-right input {
+    text-align: center;
+    font-size: 2em;
+    color: #666666;
+    margin: 50px 0 50px 0;
```
`default.css` contains rules that are potentially applied on any page, such as style of re-usable blocks. As such, it is included on *every page* and we want to keep it free of unnecessary rules. Your rule `.empty-info-text` however is only used by the page `vocabulary/add_sentences`. For such rules, we use per-page CSS file named after the page name: `webroot/css/vocabulary/add_sentences.css`. You can move it there. :slightly_smiling_face:

jiru commented:
@speedysera Thanks for the pull request. I tested it and it works great! :+1: Just some implementation details need to be fixed. Have a look at my comments. :full_moon_with_face:

AndiPersti commented on src/Template/Vocabulary/add_sentences.ctp:
```diff
@@ -27,10 +27,15 @@
-$this->set('title_for_layout', $this->Pages->formatTitle($title));
+    if (empty($langFilter)) {
+        $title = __('Vocabulary that needs sentences');
+        $this->set('title_for_layout', $this->Pages->formatTitle($title));
```
This line should be moved after the if-block. Right now the page title is only set when there is no language filter.

And I suggest to remove the indentation for the whole if-block.

speedysera replied:
nice catch!

speedysera commented:
Addressed comments and closing this branch in favor of https://github.com/Tatoeba/tatoeba2/pull/2510
@jiru @AndiPersti please take a look when you have time
